src/common/transformations/include/transformations/transpose_sinking/ts_shape_of.hpp

```diff
@@ -0,0 +1,28 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "transformations/transpose_sinking/ts_base.hpp"
+#include "transformations_visibility.hpp"
+
+namespace ov {
+namespace pass {
+namespace transpose_sinking {
+
+class TRANSFORMATIONS_API TSShapeOfForward;
```

Comment:
could you add a "backward" transformation?

Reply:
I think we don't need it.

For example if we spot ShapeOf->Gather and then TSShapeOfBackward adds a new transpose on ShapeOf's input: Transpose->ShapeOf - even we if have another transpose that could be eliminated - we wouldn't be able to do that since the first transpose has two consumers now. But even if we could, at the time TSShapeOfBackward is executed, we have no guarantee that there's another transpose that can be eliminated. So we can be left with Transpose->ShapeOf instead of ShapeOf->Gather - in general the former would be more expensive.

Reply:
what about this case? 
![image](https://github.com/openvinotoolkit/openvino/assets/21181928/e86234e1-cf85-4ad9-8273-ea06a0bc8962)